src/lib/components/composites/settings/project-settings/slr/MaybeAsDecisionSetting.svelte

```diff
@@ -0,0 +1,179 @@
+<script lang="ts">
+    import AlertDialog from "$lib/components/composites/dialog/AlertDialog.svelte";
+    import SettingsSection from "$lib/components/composites/settings/SettingsSection.svelte";
+    import Alert from "$lib/components/composites/utils/Alert.svelte";
+    import { Label } from "$lib/components/primitives/label";
+    import { Switch } from "$lib/components/primitives/switch";
+    import { maybeAsDecision } from "$lib/global-state/maybe-as-decision-state.svelte";
+    import { backendService } from "$lib/grpc-api";
+    import { Project_Settings, type Project } from "$lib/model/api/project";
+    import type { ApiError } from "$lib/model/general";
+    import { generateFieldMask } from "protobuf-fieldmask";
+    import { onMount } from "svelte";
+    import { toast } from "svelte-sonner";
+
+    interface Props {
+        projectId: string;
+        loadingProject: Promise<Project>;
+        slrSettingsLocked?: boolean;
+    }
+
+    const { projectId, loadingProject, slrSettingsLocked = false }: Props = $props();
+
+    // `isUpdatingMaybeAsDecisionSettingStatus` is initially set to `true` to disable the switch
+    let isUpdatingMaybeAsDecisionSettingStatus = $state(true);
+    let checked = $derived(maybeAsDecision.isActivated);
+
+    let isConfirmDialogOpen = $state(false);
+    let title = $state("");
+    let dialogDescription = $state("");
+    let pendingActionConfirmCallback: (() => void) | null = $state(null);
+
+    const disabled = $derived(slrSettingsLocked || isUpdatingMaybeAsDecisionSettingStatus);
+
+    let updateSLRSettingsError: ApiError | undefined = $state(undefined);
+
+    async function toggleIsMaybeAsDecisionSettingStatus(targetCheckedState: boolean) {
+        isUpdatingMaybeAsDecisionSettingStatus = true;
+
+        await loadingProject
+            .then((project) => {
+                const projectSettings = project.settings ?? Project_Settings.create();
+                projectSettings.reviewMaybeAllowed = targetCheckedState;
+                project.settings = projectSettings;
```

Comment:
```suggestion
                project.settings ??= Project_Settings.create();
                project.settings.reviewMaybeAllowed = targetCheckedState;
```
What dou you think about this. It's a bit more compact.

Reply:
You're right. This is much more convenient. I think, I needed the separate projectSettings variable for something else, but this piece of code doesn't exist any more. Therefore, the settings of the project can be updated directly.